processing/src/main/java/org/apache/druid/segment/column/TypeSignature.java

```diff
@@ -167,6 +167,12 @@ default boolean isArray()
     return getType().isArray();
   }
 
+  @JsonIgnore
+  default boolean isArrayPrimitive()
```

Comment:
Nit: "primitive array" would be the normal order of those words.  "array primitive" makes it sound like there's some primitive that is an array?